File: backend/ŠišAppApi/Models/Service.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ŠišAppApi.Models;

public class Service
{
    [Key]
    public int Id { get; set; }
    
    [Required]
    public int SalonId { get; set; }
    
    [ForeignKey("SalonId")]
    public Salon Salon { get; set; } = null!;
    
    [Required]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;
    
    public string? Description { get; set; }
    
    [Required]
    public int DurationMinutes { get; set; }
    
    [Required]
    [Column(TypeName = "decimal(10,2)")]
    public decimal Price { get; set; }
    
    public int? CategoryId { get; set; }
    
    [ForeignKey("CategoryId")]
    public ServiceCategory? Category { get; set; }
    
    public string? ImageIds { get; set; } // JSON array of image IDs
    
    public bool IsPopular { get; set; } = false;
    
    public int DisplayOrder { get; set; } = 0;
    
    public bool IsActive { get; set; } = true;
    
    [Required]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public DateTime? UpdatedAt { get; set; }
    
    public bool IsDeleted { get; set; } = false;
    
    public DateTime? DeletedAt { get; set; }
    
    // Navigation properties
    public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
    public virtual ICollection<BarberSpecialty> BarberSpecialties { get; set; } = new List<BarberSpecialty>();
    public virtual ICollection<Recommendation> Recommendations { get; set; } = new List<Recommendation>();
} 

File: sisapp_frontend/lib/models/service.dart
class Service {
  final int id;
  final int salonId;
  final int? categoryId;
  final String? categoryName;
  final String? categoryDescription;
  final String name;
  final String? description;
  final int durationMinutes;
  final double price;
  final bool isPopular;
  final bool isActive;

  Service({
    required this.id,
    required this.salonId,
    this.categoryId,
    this.categoryName,
    this.categoryDescription,
    required this.name,
    this.description,
    required this.durationMinutes,
    required this.price,
    this.isPopular = false,
    this.isActive = true,
  });

  factory Service.fromJson(Map<String, dynamic> json) {
    return Service(
      id: (json['id'] as num?)?.toInt() ?? 0,
      salonId: (json['salonId'] as num?)?.toInt() ?? 0,
      categoryId: (json['categoryId'] as num?)?.toInt(),
      categoryName: json['categoryName']?.toString(),
      categoryDescription: json['categoryDescription']?.toString(),
      name: json['name'] ?? '',
      description: json['description'],
      durationMinutes: (json['durationMinutes'] as num?)?.toInt() ?? 0,
      price: (json['price'] as num?)?.toDouble() ?? 0.0,
      isPopular: json['isPopular'] ?? false,
      isActive: json['isActive'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'salonId': salonId,
      'categoryId': categoryId,
      'categoryName': categoryName,
      'categoryDescription': categoryDescription,
      'name': name,
      'description': description,
      'durationMinutes': durationMinutes,
      'price': price,
      'isPopular': isPopular,
      'isActive': isActive,
    };
  }
  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;
    return other is Service && other.id == id;
  }

  @override
  int get hashCode => id.hashCode;
}
